### ISSUE TITLE
Export example/all.yaml

### ISSUE THREAD

**Issue by dotdoom** [dart-lang/linter#1765]
Can we have example/all.yaml exported to be used via `include:` in custom analysis_options.yaml?

Per this syntax:
https://dart.dev/guides/language/analysis-options#the-analysis-options-file

Similar to pedantic in this example:
https://github.com/dart-lang/pedantic#using-the-lints

This could be an alternative solution to #320.

**Comment by pq** [comment]
@dotdoom: out of curiosity, do you have sense for how many of `all.yaml` you'd want to exclude in practice?

**Comment by bwilkerson** [comment]
I'm not sure we want to publish that file. If we did publish it, then any rules we added to it would be a breaking change for users that have included the file.

Also, that file isn't very useful as-is because some of the rules are contradictory and therefore can't reasonably be enabled together. Users that don't realize that would likely be surprised by the requirement that any file that includes `all.yaml` need to disable some of the rules.

**Comment by dotdoom** [comment]
@pq yes, does not have to be many: https://github.com/dasfoo/delern/blob/b7fe328261340d8ff152b464cad389b4bc1ccfb2/flutter/analysis_options.yaml#L31

@bwilkerson: wouldn't the changes be versioned with linter package allowing users to adjust when they upgrade?

Indeed, the file isn't useful as-is, but it's good as a starting point and the maintenance cost seems low. It's like free tests!

**Comment by srawlins** [comment]
I second that we should not publish this file. We do not recommend that anyone try to follow all of the rules, and some conflict. It does not seem useful to me to use it as a starting point from which you would exclude the ones you don't like.

**Comment by dotdoom** [comment]
Perhaps you would consider recommending it?  For large projects it would be an overkill to follow all rules and keep fixing things, but for small ones if people want to keep their "pet project" nice and tidy, why not.

Conflicting rules aside, I find fully enabled linter quite successful at certain level in practicing software development skills in a particular framework.  You don't want annoying highlights at the beginning, nor probably when you are running 5-10 5k SLoC projects, but in between there's a group of people which I believe would benefit.

There are also quite a few things which could be linted but aren't at the moment -- making them "best-effort" until a rule is implemented in linter.  Once implemented, it will automatically catch any occurrences a human has previously missed.  I have to remember to check out the new lints every now and then to track their delivery.  Subscribing to a GitHub issue is too noisy and gives absolutely no hint at when an implemented rule arrives to the latest Flutter build.

**Comment by jamesderlin** [comment]
In my own projects, I start with `all.yaml` and manually disable the conflicting rules and other rules that I don't want. (FWIW, I currently disable 33 lints.  Of those, 9 of them are disabled because they're experimental and not recognized by Dart 2.18.) 

Every so often I download the latest version of `all.yaml`, do a file comparison to see what new lints have been added, and then merge the new lints in.  This process is rather tedious, and it's even worse because `all.yaml` includes experimental lints that won't be recognized when I run a stable version of `dart analyze`.

The process of enabling new lints would be much easier if `all.yaml` were published and versioned, also allowing it to be tied to a specific Dart SDK version.  Making experimental lints tied to development versions also would make it easier to try experimental lints while they're still experimental.

> Also, that file isn't very useful as-is because some of the rules are contradictory and therefore can't reasonably be enabled together. Users that don't realize that would likely be surprised by the requirement that any file that includes all.yaml need to disable some of the rules.

I don't think that this shouldn't be done just because it might confuse some people.
1. Someone would have to explicitly choose to consume it.
2. `all.yaml` could include a disclaimer at the top of the file discouraging people from using it.
3. Conflicting rules should be very quickly apparent when anybody tries to use it directly.
